### PR TITLE
Test active tags

### LINKED ISSUE #419: add tests for browse /tags view (chals and orgs)
Need to add tests to ensure we are displaying the categories, tags and tag-links, in the correct format for /challenges/tags and /orgs/tags view.

### PR DESCRIPTION
**Pull request by mk4111**
Closes #419 

Adds tests for /orgs/tags and /challenges/tags endpoints which both use tags-categories.html handlebar.
Checks that tags, categories and links are being displayed correctly.

### CODE REVIEW

**Comment by mk4111** [comment]
@JMurphyWeb and @SimonLab the tests are a wee repetetive but I think (until we switch them out with something more meaningful) it works... Please let me know what you think! Thanks!

File: test/runner.js [hunk 1]
```diff
@@ -1,13 +1,11 @@
 'use strict';
 
+require('./challenges/add-tags.test.js');
```
Comment (JMurphyWeb) [inline test/runner.js]: Do you think we can rename these test files to include the word 'browse' or 'explore' in them?  

I got confused earlier thinking you were talking about add tags to a challenge endpoint.

File: test/challenges/add-tags.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      }
+
+      server.inject({
+        method: 'GET',
+        url: '/challenges/tags',
+        headers: { cookie: 'token=' + primaryToken }
+      }, function (res) {
+        var unexpected = 'Please Login to view that page';
+        var expLink = '<a class="text-decoration" href="/challenges?tags=124">';
+        var expTag = '<span class="tag__label">Telecommunications</span>';
```
Comment (mk4111) [inline test/challenges/add-tags.test.js]: No that makes absolute sense :+1:

File: test/runner.js (hunk 1 above)
Comment (mk4111) [inline test/runner.js]: Absoluuutely!

**Comment by JMurphyWeb** [comment]
Looks good once test file names are changed.

**Comment by mk4111** [comment]
@SimonLab - could you review this when you have the time please? Thank you!

**Comment by SimonLab** [comment]
@JMurphyWeb conflicts to resolved with master, please reassign

**Comment by mk4111** [comment]
@SimonLab - Rebased! :taco:

File: test/challenges/select-tags-browse-view.test.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+var test = require('tape');
+var init = require('../../server/server.js');
+var config = require('../../server/config.js');
+var sessions = require('../helpers/add-sessions.js');
+// var adminToken = sessions.tokens(config.jwt_secret).admin_1;
```
Comment (SimonLab) [inline test/challenges/select-tags-browse-view.test.js]: can we delete this commented line or do you want to use it in a very near PR?

**Comment by SimonLab** [comment]
@mk4111 just one small comment about the commented line

File: test/challenges/select-tags-browse-view.test.js (hunk 3 above)
Comment (mk4111) [inline test/challenges/select-tags-browse-view.test.js]: I should actually repeat that test for admins to ensure we get the same view! I'll add that and reassign once done!

**Comment by mk4111** [comment]
@SimonLab over to you again (please!) :D :dancer:

File: test/organisations/select-tags-browse-view.test.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+var test = require('tape');
+var init = require('../../server/server.js');
+var config = require('../../server/config.js');
+var sessions = require('../helpers/add-sessions.js');
+// var adminToken = sessions.tokens(config.jwt_secret).admin_1;
```
Comment (SimonLab) [inline test/organisations/select-tags-browse-view.test.js]: @mk4111 could we delete this line also? :arrow_up:

Comment (mk4111) [inline test/organisations/select-tags-browse-view.test.js]: @SimonLab - forgot it happened twice! Sorry!

**Comment by mk4111** [comment]
@SimonLab - sorry that you're having to check over this for the second time! I've amended the other test too - hope it's ok now :wine_glass:

**Pull request merged**